src/bots/liquidator.ts

```diff
@@ -82,17 +117,197 @@ export class LiquidatorBot implements Bot {
 		return undefined;
 	}
 
+	/**
+	 * attempts to close out any open positions on this account. It starts by cancelling any open orders
+	 */
+	private async derisk() {
+		if (Atomics.compareExchange(this.deriskMutex, 0, 0, 1) === 1) {
+			return;
+		}
+
+		if (!this.SELL_OPEN_POSITIONS) {
+			return;
+		}
+
+		try {
+			const userAccount = this.clearingHouse.getUserAccount();
+			// cancel open orders
+			let canceledOrders = 0;
+			for (const order of userAccount.orders) {
+				if (order.baseAssetAmount.isZero()) {
+					continue;
+				}
+				const tx = await this.clearingHouse.cancelOrder(order.orderId);
+				logger.info(
+					`${this.name} canceling open order ${
+						order.orderId
+					} on market ${order.marketIndex.toString()}: ${tx}`
+				);
+				canceledOrders++;
+			}
+			if (canceledOrders > 0) {
+				logger.info(
+					`${this.name} canceled ${canceledOrders} open orders while derisking`
+				);
+			}
+
+			// close open orders
+			let closedPositions = 0;
+			for (const position of userAccount.positions) {
+				if (position.baseAssetAmount.isZero()) {
```

Comment:
So eventually if we have more market than user position slots, we'll need to figure out how to run the liquidator such that we can liquidate all markets. I'm wondering if we can handle this by having multiple subaccounts and each subaccount handles some subset of of markets?